scripts/tools/SetDeliveryNamespace.php

```diff
@@ -0,0 +1,124 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2021 (original work) Open Assessment Technologies SA;
+ *
+ * @author Sergei Mikhailov <sergei.mikhailov@taotesting.com>
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoDeliveryRdf\scripts\tools;
+
+use Exception;
+use InvalidArgumentException;
+use oat\oatbox\extension\script\ScriptAction;
+use oat\oatbox\reporting\Report;
+use oat\taoDeliveryRdf\model\DeliveryFactory;
+
+/**
+ * Usage:
+ *
+ * sudo -u www-data php index.php 'oat\taoDeliveryRdf\scripts\tools\SetDeliveryNamespace' -n https://taotesting.com
+ */
+class SetDeliveryNamespace extends ScriptAction
+{
+    protected function provideUsage(): array
+    {
+        return [
+            'prefix'      => 'h',
+            'longPrefix'  => 'help',
+            'description' => 'Prints this message',
+        ];
+    }
+
+    protected function provideOptions(): array
+    {
+        return [
+            'namespace' => [
+                'prefix'      => 'n',
+                'longPrefix'  => 'namespace',
+                'description' => 'A custom namespace for remotely published Delivery resources',
+            ],
+        ];
+    }
+
+    protected function run(): Report
+    {
+        if (!$this->getOption('namespace')) {
+            return $this->unsetNamespace();
+        }
+
+        $namespace = $this->getNamespace();
+
+        $deliveryFactory = $this->getDeliveryFactory();
+        $deliveryFactory->setOption(DeliveryFactory::OPTION_NAMESPACE, $namespace);
+
+        try {
+            $this->setDeliveryFactory($deliveryFactory);
+        } catch (Exception $exception) {
+            return Report::createError(
+                "Failed to set \"$namespace\" Delivery namespace.",
+                null,
+                [Report::createInfo($exception->getMessage())]
+            );
+        }
+
+        return Report::createSuccess("Registered \"$namespace\" Delivery namespace.");
+    }
+
+    protected function provideDescription(): string
+    {
+        return 'TAO DeliveryRDF - Set up remotely published Delivery resource namespace';
+    }
+
+    private function unsetNamespace(): Report
+    {
+        $deliveryFactory        = $this->getDeliveryFactory();
+        $deliveryFactoryOptions = $deliveryFactory->getOptions();
+
+        unset($deliveryFactoryOptions[DeliveryFactory::OPTION_NAMESPACE]);
+        $deliveryFactory->setOptions($deliveryFactoryOptions);
+
+        $this->setDeliveryFactory($deliveryFactory);
+
+        return Report::createSuccess('Removed a Delivery namespace.');
+    }
+
+    private function getNamespace(): string
```

Comment:
duplicated code , maybe we can reuse or turn this function in a helper of some sorts

Reply:
Introduced `DeliveryNamespaceRegistryInterface` and its implementation specifically to store and validate the overridden Delivery resource URI value.